src/IRsend.h

```diff
@@ -470,6 +470,10 @@ class IRsend {
   void sendCarrierAC40(uint64_t data, uint16_t nbits = kCarrierAc40Bits,
                        uint16_t repeat = kCarrierAc40MinRepeat);
 #endif
+#if SEND_CARRIER_AC48
+  void sendCarrier_AC48(const unsigned char data[], uint16_t nbits = kCarrierAc48Bits,
```

Comment:
please use `uint8_t` (mentoned in https://github.com/crankyoldgit/IRremoteESP8266/wiki/Library-Maintainers-Guide#c)

Reply:
Okay,  did the git reset --hard origin/master  command. 

After the git restet, 

1. do we want this to be named CarrierAC48 rather than Carrier_AC48 ?
       yes,  I made the changes to ir_Carrier.cpp, IRrecv.cpp, IRrecv.h, IRSend.cpp and IRsend.h to make this change.    I have tested the changes, and can successfully comple and run ESP8226 code.

2. please use uint8_t in scr/IRsend.h
      Is this request  to match under 64 bit codes?  

In reviewing the code,  I discovered I made an error in calling it Carrier_AC48.

by default the  Protocol  : MIDEA is tested before Protocol : Carrier_xx,  so my first attempts at running   IRreveDumpV2 returned 

Protocol  : MIDEA
Code      : 0xA10260FFFFEC (48 Bits)

When I disabled all the protocols except for Carrier I  got Protocol  : UNKNOWN  and started this fun learning exercise.

I stuck with the 48 bit idea and went from there.

should I change the name to Carrier_96 ?

I have not committed any of my changes yet, in case I need to make additional changes

Reply:
I did the rename, and uint8_t fix as well, it is an array of uint8_t

Reply:
>> by default the Protocol : MIDEA is tested before Protocol : Carrier_xx, so my first attempts at running IRreveDumpV2 returned
>> 
>> Protocol : MIDEA
Code : 0xA10260FFFFEC (48 Bits)

> In regards to your comment about MIDEA
> We will want to revisit this some more, I would like to see that the inverted part is fixed, which will make this 48bits, but you are right, in a sense this is 96 bits.

Um .. I just took a look at MIDEA. Are we sure this isn't actually the same protocol?!

@treedog56 If it detected as MIDEA, did it decode the settings correctly? And did you try controlling your AC via the MIDEA protocol at all?

Often different brands/manufacturers will use the same protocol. It could be the case here.

Reply:
this what I get from IRrecvDumpV2 with the MIDEA enabled.
Protocol  : MIDEA
Code      : 0xA10260FFFFEC (48 Bits)
 Result state is 236,255,255,96,2,161,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
Mesg Desc.: Power: Off, Mode: 2 (Auto), Celsius: Off, Temp: 17C/62F, Fan: 0 (Auto), Sleep: Off, Swing(V) Toggle: Off
uint16_t rawData[199] = {4460, 4372,  570, 1618,  572, 526,  598, 1594,  570, 528,  598, 498,  600, 498,  568, 528,  572, 1618,  570, 530,  570, 528,  572, 526,  572, 526,  570, 528,  570, 526,  596, 1596,  570, 528,  600, 500,  570, 1622,  626, 1562,  570, 528,  572, 528,  568, 528,  570, 526,  598, 500,  598, 1592,  598, 1590,  572, 1618,  570, 1616,  596, 1590,  598, 1590,  598, 1590,  598, 1592,  570, 1624,  570, 1616,  570, 1618,  570, 1618,  596, 1590,  572, 1616,  572, 1616,  572, 1618,  570, 1624,  598, 1590,  570, 1618,  570, 526,  570, 1622,  598, 1590,  596, 500,  596, 502,  510, 5282,  4444, 4376,  570, 528,  596, 1596,  570, 526,  626, 1564,  572, 1616,  598, 1590,  598, 1590,  598, 502,  570, 1620,  572, 1616,  600, 1588,  570, 1618,  570, 1616,  572, 1616,  598, 500,  598, 1596,  598, 1596,  572, 528,  596, 500,  600, 1588,  570, 1618,  570, 1616,  572, 1616,  598, 1592,  572, 528,  570, 528,  570, 526,  628, 470,  598, 502,  570, 526,  572, 526,  570, 530,  570, 530,  570, 526,  598, 500,  572, 526,  598, 498,  570, 528,  570, 528,  598, 502,  572, 528,  570, 528,  598, 500,  598, 1588,  596, 502,  596, 500,  570, 1618,  572, 1620,  510};  // MIDEA A10260FFFFEC
uint64_t data = 0xA10260FFFFEC;


I did not try sending these codes to the unit directly, but using SmartIRRepeater with MIDEA enabled  did not work.

With MIDEA  and Carrier_48 disabled I get

Protocol  : UNKNOWN
Code      : 0xA4822C4E (100 Bits)

uint16_t rawData[199] = {4376, 4450,  572, 1618,  572, 526,  596, 1594,  596, 502,  572, 526,  570, 528,  572, 526,  598, 1592,  570, 528,  598, 500,  572, 526,  572, 526,  572, 526,  572, 526,  570, 1620,  572, 528,  600, 500,  598, 1594,  572, 1616,  570, 526,  596, 502,  572, 526,  570, 526,  572, 528,  570, 1618,  598, 1590,  572, 1616,  598, 1588,  572, 1616,  572, 1616,  572, 1616,  572, 1618,  570, 1622,  572, 1616,  570, 1616,  598, 1590,  572, 1616,  596, 1590,  570, 1618,  572, 1618,  598, 1596,  598, 1588,  572, 1616,  572, 526,  570, 1620,  572, 1616,  570, 526,  572, 526,  486, 5306,  4470, 4348,  572, 526,  570, 1622,  598, 498,  572, 1620,  572, 1616,  598, 1590,  572, 1616,  570, 528,  598, 1592,  572, 1616,  598, 1590,  572, 1616,  598, 1590,  570, 1616,  572, 526,  572, 1622,  570, 1624,  570, 526,  598, 498,  572, 1616,  572, 1616,  572, 1616,  572, 1616,  572, 1618,  572, 528,  598, 500,  598, 500,  572, 526,  572, 526,  574, 524,  572, 528,  570, 530,  570, 528,  570, 528,  572, 528,  596, 500,  598, 500,  570, 526,  570, 528,  598, 502,  624, 474,  600, 498,  570, 528,  572, 1616,  572, 526,  570, 526,  598, 1590,  570, 1618,  488};  // UNKNOWN A4822C4E

After running auto_analyse_raw_data.py on the unknown code,  I made the changes to create Carrier_48 and get this.

Protocol  : CARRIER_AC48
Code      : 0xA18260FFFF6C (48 Bits)

uint16_t rawData[199] = {4458, 4390,  538, 1648,  540, 558,  540, 1652,  566, 532,  540, 556,  540, 558,  540, 558,  540, 1650,  540, 1654,  540, 558,  542, 556,  542, 556,  540, 558,  538, 560,  564, 1628,  540, 558,  540, 560,  540, 1652,  540, 1648,  540, 558,  540, 558,  540, 558,  540, 558,  540, 558,  540, 1652,  540, 1648,  540, 1648,  538, 1650,  540, 1650,  540, 1648,  538, 1648,  544, 1646,  540, 1654,  538, 1650,  540, 1646,  540, 1648,  540, 1648,  540, 1648,  566, 1622,  540, 1650,  540, 558,  540, 1648,  540, 1648,  540, 558,  570, 1622,  542, 1648,  548, 548,  540, 560,  510, 5284,  4438, 4380,  542, 556,  542, 1650,  512, 586,  538, 1654,  566, 1620,  540, 1648,  568, 1620,  538, 560,  540, 560,  542, 1650,  566, 1624,  540, 1648,  512, 1676,  540, 1648,  540, 558,  540, 1654,  566, 1628,  540, 558,  540, 558,  540, 1648,  540, 1646,  542, 1648,  540, 1648,  542, 1648,  540, 562,  564, 530,  568, 528,  542, 558,  540, 556,  540, 558,  540, 556,  538, 560,  540, 560,  540, 556,  540, 558,  542, 554,  540, 558,  568, 530,  568, 530,  538, 562,  540, 1654,  538, 558,  540, 560,  540, 1646,  542, 558,  538, 558,  540, 1648,  540, 1650,  508};  // CARRIER_AC48
uint8_t state[6] = {0xA1, 0x82, 0x60, 0xFF, 0xFF, 0x6C};

Reply:
> Protocol : MIDEA
Code : 0xA10260FFFFEC (48 Bits)
Result state is 236,255,255,96,2,161,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,0,
Mesg Desc.: Power: Off, Mode: 2 (Auto), Celsius: Off, Temp: 17C/62F, Fan: 0 (Auto), Sleep: Off, Swing(V) Toggle: Off

But is that an accurate mesg description of your remote's intended command?

Reply:
Yes, it does describe the  state of the remote.
So I retested SmartIRRepeater  

With Media enabled,  
I get  the following output.

000055.885: A 199-bit UNKNOWN message was successfully retransmitted.
but I get no response from the AC Unit.

 With Media disabled, and the Carrier_48 enabled I get 

000132.704: A 48-bit CARRIER_AC48 message was successfully retransmitted.

and the AC Unit  responds to the relayed commands.

Reply:
FYI, `SmartIRRepeater` isn't the worlds greatest software/tool/example. It isn't heavily tested. ;-)

> 000055.885: A 199-bit UNKNOWN message was successfully retransmitted.
but I get no response from the AC Unit.

Well, when it is detecting it as `UNKNOWN`, SmartIRRepeater will just send the Raw capture back. Which, in theory is pretty much the most accurate it can be. (There are limitations but we don't need to go into it here)

It **should** be detecting it as MIDEA anyway, so something is _up_!

FWIW, `199` is the correct length for a MIDEA message in Raw Data.

> Yes, it does describe the state of the remote.

**This** is _really_ strong evidence that they are the _same_ protocol.
It could be subtle timing differences when sending etc that might be causing problems.

I'm also wondering if we've screwed up his build environment.

Can you try this example code, and modify it for use with Midea?
https://github.com/crankyoldgit/IRremoteESP8266/blob/master/examples/TurnOnArgoAC/TurnOnArgoAC.ino

e.g.
```
#include <Arduino.h>
#include <IRremoteESP8266.h>
#include <IRsend.h>
#include <ir_Midea.h>

const uint16_t kIrLed = 4;  // ESP8266 GPIO pin to use. Recommended: 4 (D2).
IRMideaAC ac(kIrLed);  // Set the GPIO to be used to sending the message.

void setup() {
  ac.begin();
  Serial.begin(115200);
}

void loop() {
  Serial.println("Sending...");

  // Set up what we want to send. See ir_Midea.cpp for all the options.
  ac.setPower(true);
  ac.setMode(kMideaACAuto);
  ac.setFan(kMideaACFanAuto);
  ac.setTemp(62);  // Fahrenheit

#if SEND_MIDEA
  // Now send the IR signal.
  ac.send();
#else  // SEND_MIDEA
  Serial.println("Can't send because SEND_MIDEA has been disabled.");
#endif  // SEND_MIDEA

  delay(5000);
}
```